doc/formats/morphology.rst

```diff
@@ -0,0 +1,142 @@
+* Check that the parent id of a sample refers to an existing sample.
+
+In addition, all interpretations agree that a *segment* is (in the common case) constructed between a sample and
+its parent and inherits the tag of the sample; and if more than 1 sample have the same parent, the parent sample
+is interpreted as a fork point in the morphology, and acts as the proximal point to a new branch for each of its
+"child" samples. There a couple of exceptions to these rules which are listed below.
+
+Arbor interpretation:
+"""""""""""""""""""""
+In addition to the previously listed checks, the arbor interpretation explicitly disallows SWC files where the soma is
+described by a single sample. It constructs the soma from 2 or more samples, forming 1 or more segments. A *segment* is
+always constructed between a sample and its parent. This means that there are no gaps in the resulting morphology.
+
+Arbor has no magic rules or transformations for the soma. It can be a single branch or multiple branches; segments
+of a different tag can connect to its distal end, proximal end or anywhere in the middle. For example, to create a
+morphology with a single segment soma; a single segment axon connected to one end of the soma; and a single segment
+dendrite connected to the other end of the soma, the following swc file can be used:
+
+
+.. literalinclude :: example.swc
+   :language: python
+   :linenos:
+
+Samples 1 and 2 will form the soma; samples 1 and 3 will form the axon, connected to the soma at the proximal end;
+samples 2 and 4 will form the dendrite, connected to the soma at the distal end. The morphology will look something
+like this:
+
+.. figure:: ../gen-images/swc_morph.svg
+   :width: 400
+   :align: center
+
+
+Allen interpretation:
+"""""""""""""""""""""
+In addition to the previously mentioned checks, the Allen interpretation expects a single-sample soma to be the first
+sample of the file and to be interpreted as a spherical soma. Arbor represents the spherical soma as a cylinder with
+length and diameter equal to the diameter of the sample representing the sphere.
+
+This interpretation also expects that samples have the same tag as their parent samples, with the exception of samples
+that have the soma sample as a parent. In this case, when a sample's parent is the soma, no *segment* is created
+between the 2 samples; instead there is a gap in the morphology (represented electrically as a zero-resistance wire).
+Samples with the soma as a parent start new segments, that connect to the distal end of the soma if they are dendrites,
+or to the proximal end of the soma if they are axons or apical dendrites. Only axons, dendrites and apical dendrites
+(tags 2, 3 and 4 respectively) are allowed in this interpretation, in addition to the spherical soma.
+
+Finally the Allen institute interpretation of SWC files centres the morphology around the soma at the origin (0, 0, 0)
+and all samples are translated in space towards the origin.
+
```

Comment:
Would the difference in interpretation actually show up in a way visible in the renderings? If so, maybe add a three way comparison.

Reply:
Some of it would, some wouldn't. The main difference is in what can/can't be represented.
We can add some form of visual representation of the difference, but I'd prefer to keep it out of this PR. As I mentioned, this file is just copied from elsewhere in the code, so none of this file is new.

Reply:
Sure thing, that was mainly my curiosity, but let's keep it in mind for another pass on the docs.